src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
```

Comment:
Don't add empty parameter comments

```suggestion
     */
```

Reply:
ok, removed. But I am editing my working branch, and will copy then push.
To avoid confusion I did not click on the "commit suggestion button"